### PR TITLE
AEA-770 Progressing the Identity scenario

### PR DESCRIPTION
**Pull request by 5A11**
This PR adds code that progresses the identity demo's scenario.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/aries-cloud-agent-demo.md [hunk 1]
```diff
@@ -90,17 +99,35 @@ At this point, the two ACAs are connected to each other.
...
 
 ``` bash
 pip install aries-cloudagent
 ```
 
+This demo has been successfully tested with aca-py version 0.4.5.
+
+This demo requires an instance of von network running in docker locally (for more info see <a href="https://github.com/bcgov/von-network#running-the-network-locally" target=_blank>here</a>)
+
+This demo has been successfully tested with the von-network git repository pulled on 07 Aug 2020.
```
Comment (DavidMinarsch) [inline docs/aries-cloud-agent-demo.md:114]: You should point to a specific commit or tag (ideally have the full command to clone the repo in the correct state)

File: docs/aries-cloud-agent-demo.md [hunk 2]
```diff
@@ -90,17 +99,35 @@ At this point, the two ACAs are connected to each other.
...
+
+In the first terminal move to the `von-network` directory and run an instance of `von-network` locally in docker.
+
+This <a href="https://github.com/bcgov/von-network#running-the-network-locally" target=_blank>tutorial</a> has information on starting (and stopping) the network locally.
+
+``` bash
+./manage build
+./manage start --logs
+``` 
+Once the ledger is running, you can see the ledger by going to the web server running on port 9000. On localhost, that means going to <a href="http://localhost:9000" target=_blank>http://localhost:9000</a>.
```
Comment (DavidMinarsch) [inline docs/aries-cloud-agent-demo.md:130]: Please make clear that this is an optional part of the demo

File: packages/fetchai/skills/aries_alice/handlers.py [hunk 3]
```diff
@@ -84,6 +87,38 @@ def _admin_post(self, path: str, content: Dict = None):
             context=EnvelopeContext(connection_id=HTTP_CLIENT_CONNECTION_PUBLIC_ID),
         )
 
+    def _handle_received_invite(self, invite_detail: Dict):
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/aries_alice/handlers.py:90]: Missing docstring and return type hint

File: packages/fetchai/skills/aries_faber/handlers.py [hunk 4]
```diff
@@ -47,6 +48,8 @@
 ADMIN_COMMAND_CREATE_INVITATION = "/connections/create-invitation"
 ADMIN_COMMAND_STATUS = "/status"
 HTTP_COUNTERPARTY = "HTTP Server"
+FABER_ACA_IDENTITY = "Faber_ACA"
+LEDGER_URL = "http://127.0.0.1:9000"
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/aries_faber/handlers.py:52]: These should all be "default" constants, and then be doubled by values settable via the yaml config file. So you can configure it with alternative values (see our other demos, in particular the strategy.py files for examples of what I mean)

File: packages/fetchai/skills/aries_faber/handlers.py [hunk 5]
```diff
@@ -58,9 +61,22 @@ def __init__(self, **kwargs):
         """Initialize the handler."""
         super().__init__(**kwargs)
 
+        # ACA stuff
+        self.aca_identity = FABER_ACA_IDENTITY
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/aries_faber/handlers.py:65]:
```suggestion
        self.aca_identity = kwargs.pop("faber_aca_identity", FABER_ACA_IDENTITY)
```
This makes it configurable via yaml. file

File: packages/fetchai/skills/aries_faber/handlers.py [hunk 6]
```diff
@@ -92,6 +108,25 @@ def _admin_post(self, path: str, content: Dict = None) -> None:
         ), "faber -> http_handler -> _admin_post(): something went wrong when sending a HTTP message."
         self.context.outbox.put_message(message=request_http_message)
 
+    def _post(self, path: str, content: Dict = None) -> None:
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/aries_faber/handlers.py:111]: In complete doc strings

Comment (5A11) [reply packages/fetchai/skills/aries_faber/handlers.py:111]: Done

File: packages/fetchai/skills/aries_faber/handlers.py (hunk 5 above)
Comment (5A11) [reply packages/fetchai/skills/aries_faber/handlers.py:65]: Done

File: packages/fetchai/skills/aries_faber/handlers.py (hunk 4 above)
Comment (5A11) [reply packages/fetchai/skills/aries_faber/handlers.py:52]: Done

File: packages/fetchai/skills/aries_alice/handlers.py (hunk 3 above)
Comment (5A11) [reply packages/fetchai/skills/aries_alice/handlers.py:90]: Done

File: docs/aries-cloud-agent-demo.md (hunk 1 above)
Comment (5A11) [reply docs/aries-cloud-agent-demo.md:114]: Done

File: docs/aries-cloud-agent-demo.md (hunk 2 above)
Comment (5A11) [reply docs/aries-cloud-agent-demo.md:130]: Without the ledger it wouldn't really be an identity demo. The AEAs connect with each other and connect with their respective ACAs and then do some basic communication framework side to find each other and send some data around. They wouldn't be able to achieve any identity interaction, unique to this demo. 

Even making ACAs introduced and connected to one another (in the Aries sense) requires Faber to have registered a DID on the ledger so Alice can pick that up.

**Pull request merged**